json/src/test/java/org/teavm/flavour/json/test/DeserializerTest.java

```diff
@@ -408,6 +414,7 @@ public String getBar() {
     @JsonTypeInfo(use = Id.NAME, include = As.WRAPPER_OBJECT)
     @JsonTypeName("base")
     @JsonSubTypes({ @Type(InheritanceAsWrapperObject.class) })
+    @JsonAutoDetect(fieldVisibility = Visibility.PUBLIC_ONLY)
```

Comment:
You should not have changed existing tests, since they are correct. They fail due to another bug, see ClassInformation.java:35 (should be PUBLIC_ONLY).

Reply:
That should be fixed with last commit amend